front/app/modules/commercial/customizable_homepage_banner/admin/CTASettings/CustomizedButtonSettings.tsx

```diff
@@ -81,46 +81,60 @@ const getUrlErrors = (
   return null;
 };
 
+type SettingKeyType =
+  | 'banner_cta_signed_out_text_multiloc'
+  | 'banner_cta_signed_in_text_multiloc'
+  | 'banner_cta_signed_out_url'
+  | 'banner_cta_signed_in_url';
+
 interface Props {
+  buttonUrl: string;
+  buttonMultiloc: Multiloc;
   buttonConfig?: CustomizedButtonConfig;
-  handleSettingOnChange: (settingKey: string, settingValue: any) => void;
+  handleSettingOnChange: (
+    settingKey: SettingKeyType,
+    settingValue: Multiloc | String
+  ) => void;
   signInStatus: 'signed_out' | 'signed_in';
   errors: CLErrors;
   className?: string;
 }
 
 const CustomizedButtonSettings = ({
-  buttonConfig,
+  buttonMultiloc,
+  buttonUrl,
   handleSettingOnChange,
   signInStatus,
   errors,
   className,
   intl: { formatMessage },
 }: Props & InjectedIntlProps) => {
-  const customizedButtonKey = `cta_${signInStatus}_customized_button`;
-  const handleOnChange =
-    (buttonKey: keyof CustomizedButtonConfig) => (value: any) => {
-      handleSettingOnChange(customizedButtonKey, {
-        ...buttonConfig,
-        [buttonKey]: value,
-      });
-    };
   const handleTextOnChange = (textMultiloc: Multiloc) => {
-    handleOnChange('text')(textMultiloc);
+    const signedInStatusKey =
+      signInStatus === 'signed_out'
+        ? 'banner_cta_signed_out_text_multiloc'
+        : 'banner_cta_signed_in_text_multiloc';
+    handleSettingOnChange(signedInStatusKey, textMultiloc);
     // We set it to {} because if it's set to the current error and there are two empty locale inputs,
```

Comment:
This is the weird validation I mentioned, the error listed here has returned (if two locales are empty, typing one character in the first moves you to the second). I'm looking into this but I suspect it can be simplified. If the rest of the PR is good then I would suggest merging to the feature branch and adding a separate task to refactor this

Reply:
Donezo: https://citizenlab.atlassian.net/browse/CL-1135